### DISCUSSION TITLE
Enable NESD when using ionospheric estimation in topsStack

### DISCUSSION THREAD

**Discussion by bjmarfito** [isce-framework/isce2#705] [Q&A] [answered]
Hi @CunrenLiang,

I would like to ask if it is recommended to enable NESD and ionospheric estimation when running stackSentinel.py. I am asking this since in topsApp.py ESD is recommended to be turned off when ionospheric phase is being estimated.

Best,
Bryan

**Comment by CunrenLiang** [comment, marked as answer]
Bryan,

In topsApp.py, there is no need for ESD if you choose to do ionosphere estimation, since it is included in ionosphere estimation. You can do ESD though, it is just a waste of time.

In topsStack, NESD is not tied to ionosphere estimation in current version. You don't have to make your choice by considering ionopshere estimation. However, in the updated version (pull request https://github.com/isce-framework/isce2/pull/600), if you want to remove azimuth burst ramps caused by ionosphere in the ionospheric correction, you need to turn NESD on. It is related to the following ionosphere estimation.

Cunren

**Reply by bjmarfito** [reply]
Thank you Cunren! I hope that the new version of the ionospheric correction will be merged soon.

**Reply by mgovorcin** [reply]
it appears to me ESD is computed with ionosphere estimation in topsApp only when using 'considerBurstProperties' (which is off by default). Maybe I missed something, @CunrenLiang, is this right?

**Reply by CunrenLiang** [reply]
that's right.
